compiler/AST/expr.cpp

```diff
@@ -4021,6 +4048,38 @@ GenRet CallExpr::codegen() {
         bool handled = true;
         switch (call->primitive->tag)
         {
+         case PRIM_GET_REAL:
+         case PRIM_GET_IMAG:
+         {
+           bool isReal = call->primitive->tag == PRIM_GET_REAL;
+           if (call->get(1)->typeInfo()->symbol->hasFlag(FLAG_WIDE_REF)) {
+             // move(wide_real, prim_get_real(wide_complex));
+             // turns into: wide_real.locale = wide_complex.locale;
+             //             wide_real.addr = prim_get_real(wide_complex.addr);
+             GenRet t1 = createTempVar(call->get(1)->typeInfo()->getRefType());
+             Type* cplxType = call->get(1)->typeInfo()->getRefType();
+             codegenAssign(t1, codegenRaddr(call->get(1)));
+             GenRet t2 = createTempVar(get(1)->typeInfo()->getRefType());
```

Comment:
You might want to store the ref type information in a separate variable, since you access it a couple of times.  Not essential, though